premium/usage.go

```diff
@@ -240,6 +263,35 @@ func NewUsageClient(meta plugin.Meta, ops ...UsageClientOptions) (UsageClient, e
 	return u, nil
 }
 
+func (u *BatchUpdater) setupAWSMarketplace() error {
+	cfg, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to load AWS config: %w", err)
+	}
+	// This allows us to be able to inject a mock client for testing
+	if u.awsMarketplaceClient == nil {
+		u.awsMarketplaceClient = marketplacemetering.NewFromConfig(cfg)
+	}
+	u.teamName = "AWS_MARKETPLACE"
+	// This needs to be larger than normal, because we can only send a single usage record per second (from each compute node)
+	u.batchLimit = 1000000000
+
+	u.minTimeBetweenFlushes = 1 * time.Minute
```

Comment:
Thinking about it now, I think a better solution than `u.batchLimit = 1000000000` would be to add a parameter like `minTimeSinceLastFlush`, set to `1s`, and then enforce this in the Close function, which should sleep until the min time threshold is reached.

This is non-blocking, but maybe we can open an issue for this kind of improvement

Reply:
https://github.com/cloudquery/cloudquery-issues/issues/2177